archive/oauth-and-openid.md

```diff
@@ -57,17 +57,88 @@ With this process
 * Client doesn't need to request with Resource Owner's credential.
 * Resource Server doesn't need to authenticate user always.
 
+# Obtaining Authorization
 
-## OAuth 2.1 does not recommend Implicit Grant
+## Authorization Code Grant
 
-OAuth 2.0 supports [Implicit Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-1.3.2), however,
-OAuth 2.1 doesn't recommend it but recommend trying authorization request with
-[PKCE](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-01#oauth-2-0-differences). Moreoever, 
-there are other considerations in OAuth flow.  OAuth 2.1 already expired, however, those approached are problem 
-to think about when providing OAuth flow.
+This grant type is used to obtain access token and refresh token. It's important to create redirection-based flow
+with confidential client.
```

Comment:
How about removing this newline? I guess this is usable for IDE, but I recommand to use the setting the max character of a line. The other lines too.

Reply:
I intended that the below thing. This will be represented as a single-line in markdown viewer, but it can be weird in the editor. This will help us to see this file in other environments.

- AS-IS
```
This grant type is used to obtain access token and refresh token. It's important to create redirection-based flow**\n\r**
with confidential client.
```

- TO-BE
```
This grant type is used to obtain access token and refresh token. It's important to create redirection-based flow with confidential client.
```

Reply:
Oh, which viewer are you use? 
If I want to add newline it in markdown, I need to add two new lines, so I thought the current document does not break the format intended by markdown. And I don't think that even if I write it in a form that does not exceed 112 which is the code limit length of my local environment, it will affect other people's environment.

Reply:
The viewer does not have a problem, but the editor may have a problem. If someone wish to see this content in the editor, the line can be shown differently because of the window width. The below is an example using the chrome with raw file. The reader will expect that the newline will be shown only if the newline is needed. But AS-IS content has a newline caused by the limit length of an environment. I think removing such newlines is better to the reader using editors like vscode, IntelliJ.

![image](https://github.com/user-attachments/assets/3cd1a3f6-eb03-4044-b4f6-e806799506f4)
![image](https://github.com/user-attachments/assets/db99db09-c139-4f0c-8395-de407c5a3d06)

Reply:
The above reader will expect the below.

"This grant type is used to obtain access token and refresh token. It's \n
important to create redirection-based flow with confidential client." (newline caused by window width)

The below reader will expect the below.

"This grant type is used to obtain access token and refresh token. It's important to create redirection-based flow with confidential client." (no newlines)

Reply:
I think the code editor has a functionlity to fit the sentence in a window. How about using this when writing the markdown files?

Reply:
I also use IntelliJ. But, my Intellij does not give me the text `/n` for the new line.
I expect that the reader should render view following convention of git markdos, but some of them are not work as it.
I've written various marksdown, but I've never been asked to do anything like this. I've been uncomfortable with the markdown written by someone else because the newline is taken from a different location than I expected, but I thought it was following the required convention and I've adapted to it.

So what I want to say is that I want to make our convention. Do you want to get rid of newline? Actually, I like having newline.